### PR TITLE
feat: core: Switch PrePOMarketFactory to SafeAccessControlEnumerableUpgradeable

### PR DESCRIPTION
**Pull request by davidprepo**
Spec: https://www.notion.so/prepo/SC-Backlog-Tasks-0c6725bb870f4c22b7ba771347333662#84f44167cf584145833e79dcf9319f99

- Add role access control to `PrePOMarketFactory`
- Updated `PrePOMarket` test to grant required roles
- Additional refactor of role utils to allow code reuse when testing roles in a more complex transaction setup
- Cleanup: Removing unused variables `user2` and `TEST_REDEMPTION_FEE`

### CODE REVIEW

File: apps/smart-contracts/core/contracts/PrePOMarketFactory.sol [hunk 1]
```diff
@@ -5,20 +5,25 @@ import "./LongShortToken.sol";
...
-  OwnableUpgradeable,
-  ReentrancyGuardUpgradeable
+  ReentrancyGuardUpgradeable,
+  SafeAccessControlEnumerableUpgradeable
 {
   mapping(address => bool) private validCollateral;
   mapping(bytes32 => address) private deployedMarkets;
 
+  bytes32 public constant CREATE_MARKET_ROLE =
+    keccak256("PrePOMarketFactory_createMarket()");
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/contracts/PrePOMarketFactory.sol:21]:
```suggestion
    keccak256("createMarket");
```
As per discord discussion

File: apps/smart-contracts/core/test/PrePOMarketFactory.test.ts [hunk 2]
```diff
@@ -22,36 +29,49 @@ describe('=> PrePOMarketFactory', () => {
...
 
   describe('# initialize', () => {
-    it('owner should be set to deployer', async () => {
-      expect(await prePOMarketFactory.owner()).to.eq(deployer.address)
+    it('sets role constants to the correct hash', async () => {
+      await testRoleConstants([
+        prePOMarketFactory.CREATE_MARKET_ROLE(),
+        'PrePOMarketFactory_createMarket()',
+        prePOMarketFactory.SET_COLLATERAL_VALIDITY_ROLE(),
+        'PrePOMarketFactory_setCollateralValidity()',
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/test/PrePOMarketFactory.test.ts:60]:
```suggestion
        'setCollateralValidity',
```

**Review by ramenforbreakfast — APPROVED** [review]
LGTM besides concern with additional utility created for the testing corner case.

**Review by XavierEkkel — APPROVED** [review]
LGTM after @ramenforbreakfast's comments

File: apps/smart-contracts/core/test/utils.ts [hunk 3]
```diff
@@ -145,6 +145,17 @@ export async function testRoleConstants(roles: (Promise<string> | string)[]): Pr
   await Promise.all(tests)
 }
 
+export async function expectMissingRole(
```
[the remark this answers is not in the document]
Comment (davidprepo) [reply apps/smart-contracts/core/test/utils.ts:148]: Agree, will change.
